### PR TITLE
orchestrator-kubernetes: Remove TODO about owner_references

### PR DESCRIPTION
**Pull request by def-**
I'm not sure if more has to be done to actually set the correct values. @pH14 @benesch Can you please check?

Follow-up to #29614 
### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] If this PR includes major [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note), I have pinged the relevant PM to schedule a changelog post.

### CODE REVIEW

**Comment by benesch** [comment]
Hmm. I don't think "no owner references" is what the TODO intended. I assume the "correct" owner reference is the `Environment` resource. But that's actually itself a design flaw, because we want the VPC endpoints to persist even if the environment is deprovisioned/reprovisioned, or migrated between Kubernetes clusters.

Also, the owner references here also don't really matter, because the VPC endpoints get created in the environment namespace, and everything inside that namespace gets deleted when the namespace gets deleted.

I'm inclined to just delete the TODO and move on. @pH14 or @alex-hunt-materialize team would know better if there's some reason it's important to have an owner reference here, vs just relying on namespace deletion.

**Comment by alex-hunt-materialize** [comment]
For now, we should just leave it without owner references, and let the namespace deletion clean them up.

Long-term, we want to move these outside the individual environment clusters, so they may persist for the lifetime of the EnvironmentAssignment, not of the Environment. That would allow us to soft-delete and restore, and to migrate across environment clusters without breaking their privatelink.

File: src/orchestrator-kubernetes/src/cloud_resource_controller.rs [hunk 1]
```diff
@@ -50,8 +50,7 @@ impl CloudResourceController for KubernetesOrchestrator {
                 labels: Some(labels),
                 name: Some(name.clone()),
                 namespace: Some(self.kubernetes_namespace.clone()),
-                // TODO owner references #28729
-                //owner_references: todo!(),
+                owner_references: vec![].into(),
```
Comment (benesch) [inline src/orchestrator-kubernetes/src/cloud_resource_controller.rs:53]:
```suggestion
```

**Review by benesch — COMMENTED** [review]
Great—@def- I think we can just eliminate the TODO and the commented out code entirely then.

**Review by benesch — APPROVED** [review]
Thank you!

**Pull request merged**